purchase_request/models/purchase_request.py

```diff
@@ -160,6 +174,18 @@ def _compute_is_editable(self):
         store=True,
     )
 
+    @api.depends("approver_user_id")
```

Comment:
Add the state so it trigger also when the state changes
```suggestion
    @api.depends("approver_user_id", "state")
```